apollo/lib/schemas/user.py

```diff
@@ -25,13 +32,43 @@ def username_must_be_unique(cls, value, **kwargs):
     @validator('password')
     @classmethod
     def no_whitespace_in_password(cls, value):
-        if ' ' in value:
-            raise ValueError("password can't contain whitespaces")
-
-        return value
+        return check_for_whitespace(value)
 
 
 class UserSchema(ORMBase):
     id: uuid.UUID
     username: str
+    has_changed_initial_password: bool
     role: Optional[RoleSchema]
+
+
+class UpdateUserSchema(BaseModel):
+    old_password: constr(min_length=8, strip_whitespace=True)
+    password: constr(min_length=8, strip_whitespace=True)
+    password_confirm: constr(min_length=8, strip_whitespace=True)
+
+    @validator('password')
+    @classmethod
+    def no_whitespace_in_password(cls, value):
+        return check_for_whitespace(value)
+
+    @validator('password_confirm')
+    @classmethod
+    def password_must_match(cls, v, values):
+        if v != values.get('password'):
+            raise ValueError('passwords must match')
+
+        return v
+
+    @validator('password')
+    @classmethod
+    def password_cannot_not_match_old_password(cls, v, values):
+        if not values.get('old_password'):
```

Comment:
I'd add this to the validation of `old_password` instead, I think you'd expect it to be an error on that key, also for front-end sake.

Optionally, this also allows to refer to `old_password` as `old password` in your actual error message which might make it slightly easier to display the error to an end user.

Reply:
I removed it for now, because all the fields are required atm, seems silly to validate if a field is present.